File: POO_16/Ficha 2/FX2_EX2_2/Tabela.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Tabela.cpp
 * Author: costa
 * 
 * Created on 4 de Outubro de 2016, 22:32
 */

#include "Tabela.h"

#include <cstdlib>
#include <iostream>
#include <string>
#include <cstring>
#include "Tabela.h"

using namespace std;
 
 
Tabela::Tabela(int num){
    inicializa(num);
}

void Tabela::inicializa(int num){
    n = DIM;
    for (int i = 0 ; i < DIM; i++) a[i] =  num;
}

void Tabela::listar(){
    cout << "\nListar Estrutura: ";
    for(int i = 0; i < n; i++)
        cout <<"\n a[" << i << "]" << a[i];
    cout << endl;
}

int Tabela::getA(int indice){
    if(indice < 0 || indice >= n) return 0;
    
    return a[indice];
}


void Tabela::setA(int indice, int valor) {
    if(indice < 0 || indice >= n) return;
    a[indice] = valor;
}

int & Tabela::elementoEm(int indice){
    static int lixo = 0;
    if(indice < 0 || indice >= n) return lixo;
    return a[indice];
    
}

File: PA/FX1_EX4/src/fx1_ex4/Tabela.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package fx1_ex4;

/**
 *
 * @author costa
 */
class Tabela {
    
    private static final int DIM = 20;
    private int [] a = new int[DIM];
    private int duplicados = 0;
    
    public Tabela(){
        preenche();
    }
    
    public void preenche(){
        
        duplicados = 0; 
        for (int i = 0 ; i < a.length ; i++)
        {
            int num;
            int vezes = 0;
            do{
                num = (int) (Math.random() * 101);
                vezes++;
                
            }while(pesquisa(num,i));
            
            if(vezes > 1)
            {
                duplicados++;
            }
            
            a[i] = num;
        }
    }
    
    private boolean pesquisa(int valor, int quantos)
    {
        for(int i = 0 ; i < quantos ; i++)
        {
            if(a[i] == valor)
                return true;
        }
        return false;
    }
    
    
    @Override
    public String toString(){
        
        if(a == null)
        {
            return "\narray vazio";
        }
        String s = "";
        
        for(int i = 0 ; i < a.length ; i++)
        {
            s += " \n a[" + i + "] = " + a[i];
        }
        s+= "\n duplicados " + duplicados + "\n";
        
        return s;
    }
    
}
